src/util/address.rs

```diff
@@ -242,7 +248,10 @@ impl Display for Address {
         match self.payload {
             // note: serialization for pay-to-pk is defined, but is irreversible
             Payload::Pubkey(ref pk) => {
-                let hash = &Hash160::from_data(&pk.serialize_uncompressed()[..]);
+                let mut bytes = Vec::new();
+                pk.write_into(&mut bytes);
+
+                let hash = &Hash160::from_data(&bytes);
```

Comment:
The "write to `Vec` then hash" seems to be a recurring pattern, maybe there should be a function that can feed any consensus encodable structure into a hash function (maybe this even exists already, idk)?

EDIT: this seems to be easier to fix once #215 gets merged, maybe just add a `TODO`?

Reply:
Added a `TODO`!